src/tests/ftest/container/query_properties.py

```diff
@@ -0,0 +1,140 @@
+            srv_verify,
+            chksum_type,
+            chunk_size
+        ]
+
+        # Create container with the DaosContProperties.
+        self.container.create(con_in=con_in)
+
+        # Open the container.
+        self.container.open(pool_handle=self.pool.pool.handle)
+
+        # Prepare the DaosProperty data structure that stores the values that are
+        # configured based on the properties we used during create. Here, we create
+        # the empty data structure and set the dpe_type fields. The values (dpe_val) will
+        # be filled during query. See DaosContainer.create() and DaosContProperties for
+        # more details.
+
+        # If DaosContProperties.type is not "Unknown", there will be 4 elements.
+
+        # Element 0: Layout type, which is determined by the container type. If the
+        # container type is POSIX, we expect the value to be
+        # DaosContPropEnum.DAOS_PROP_CO_LAYOUT_POSIX.value
+
+        # Note: enable_chksum needs to be set to True to get the following 3 elements.
+        # Element 1: Checksum. In default, we expect it to be 1.
+        # Element 2: Checksum server verify. Since we updated the srv_verify to True, we
+        # expect the value to be 1.
+        # Element 3: Checksum chunk size. In default we expect it to be 16384.
+        daos_property = daos_cref.DaosProperty(4)
+
+        daos_property.dpp_entries[0].dpe_type = ctypes.c_uint32(
+            DaosContPropEnum.DAOS_PROP_CO_LAYOUT_TYPE.value)
+        daos_property.dpp_entries[1].dpe_type = ctypes.c_uint32(
+            DaosContPropEnum.DAOS_PROP_CO_CSUM.value)
+        daos_property.dpp_entries[2].dpe_type = ctypes.c_uint32(
+            DaosContPropEnum.DAOS_PROP_CO_CSUM_SERVER_VERIFY.value)
+        daos_property.dpp_entries[3].dpe_type = ctypes.c_uint32(
+            DaosContPropEnum.DAOS_PROP_CO_CSUM_CHUNK_SIZE.value)
+
+        try:
+            cont_info = self.container.container.query(daos_property=daos_property)
+        except DaosApiError as error:
+            self.log.info("Container query error! %s", error)
+
+        # Sanity check that query isn't broken.
+        uuid = conversion.c_uuid_to_str(cont_info.ci_uuid)
+        self.log.info("UUID obtained from query = %s", uuid)
+        if uuid != self.container.container.get_uuid_str():
+            msg = ("Container UUID obtained after create and after query don't match! "
+                   "Create: {}, Query: {}".format(
+                       self.container.container.get_uuid_str(), uuid))
+            errors.append(msg)
+
+        # Verify values set in daos_property.
+        # Verify layout type.
+        layout_type = daos_property.dpp_entries[0].dpe_val
+        if layout_type == DaosContPropEnum.DAOS_PROP_CO_LAYOUT_POSIX.value:
+            self.log.info("Layout type is POSIX as expected.")
+        else:
+            self.log.info("Layout type is NOT POSIX!")
+            errors.append("Layout type is not POSIX! %s", layout_type)
```

Comment:
Any specific reason why we call `self.log.info` for layout type but not the other properties below?

Reply:
No, these are just for debugging.